### PR TITLE
Respect allowed types in paste

### PR DESCRIPTION
**Pull request by cekk**
See [#3267](https://github.com/plone/Products.CMFPlone/issues/3267)

### CODE REVIEW

**Comment by cekk** [comment]
@jenkins-plone-org please run jobs

**Comment by mamico** [comment]
@jenkins-plone-org please run jobs

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

**Comment by mamico** [comment]
@jenkins-plone-org please run jobs

**Comment by jensens** [comment]
The test failure is valid. Before we check the constraints, it need to be checked if the pasted object is [`IContentish`](https://cmfcore.readthedocs.io/en/latest/api/content.html#Products.CMFCore.interfaces.IContentish). Only then we have to check the constraints.

We always allow paste of non contentish items, like in this test, the `portal_catalog`. But there may be other non-content objects anywhere in the structure as some add-ons may need them.

https://github.com/plone/Products.CMFPlone/blob/d3ea57f5f950d266f5c0838781010e9d65d90ced/Products/CMFPlone/tests/testCatalogTool.py#L983-L989

**Comment by mamico** [comment]
@jenkins-plone-org please run jobs

**Review by jensens — APPROVED** [review]

**Comment by jensens** [comment]
this needs a manual rebase and can be merged if all is green again.

**Comment by jensens** [comment]
@mamico thanks for taking care of this one

**Comment by mamico** [comment]
@jenkins-plone-org please run jobs

**Comment by mamico** [comment]
@jenkins-plone-org please run jobs

**Review by mauritsvanrees — APPROVED** [review]
LGTM. Thanks!

File: news/146.bug [hunk 1]
```diff
@@ -0,0 +1 @@
+Respect locally allowed types when pasting objects [cekk]
```
Comment (gforcada) [inline news/146.bug:1]: Late on the party, but this suffix `.bug` will not work with `towncrier` expectations, see https://github.com/plone/plone.dexterity/blob/master/pyproject.toml it should be `bugfix`

I will change it directly on master branch.

**Pull request merged**